server/src/test/java/org/elasticsearch/common/RoundingWireTests.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common;
+
+import org.elasticsearch.common.Rounding.DateTimeUnit;
+import org.elasticsearch.common.io.stream.Writeable.Reader;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.test.AbstractWireSerializingTestCase;
+
+public class RoundingWireTests extends AbstractWireSerializingTestCase<Rounding> {
```

Comment:
I'm a nervous nellie... would it be possible to explicitly test the "short" deprecated TZs from DateUtils, just to make sure we aren't breaking them?

Reply:
So I'm not 100% sure I can or should. Like, I can write a test that uses one of the short IDs, but I'm not super familiar with how to use them properly. The randomized testing doesn't like to generate them, presumably because they don't work on all jvms. I know one of them didn't work for me locally.

The class itself only takes ZoneId, not string. So it'd be fairly difficult to get one of those deprecated zone ids into it.

Reply:
Ah, I see.  And tracing backwards, DateHisto parses the input string with `ZoneId.of()` which should reject these deprecated short IDs as well.

And the only other potential problem area is Rollup (which stored some short TZs before the javatime switch), but we have an alias lookup map for that use-case.

Righto, carry on :+1: